src/pybind/mgr/cephadm/services/cephadmservice.py

```diff
@@ -994,7 +996,8 @@ def generate_config(self, daemon_spec: CephadmDaemonSpec) -> Tuple[Dict[str, Any
         config = {
             "crt": cfg.crt,
             "key": cfg.key,
-            "token": cfg.token
+            "token": cfg.token,
+            'binary': self.mgr.cephadm_binary_path,
```

Comment:
I could be off, but it looks like we pass the config dict into the CephadmDaemon deploy_daemon_unit function 

https://github.com/ceph/ceph/blob/4166eca33dcb242f5f60262033737867ec8fde59/src/cephadm/cephadm#L2497-L2504

which then writes all these files out

https://github.com/ceph/ceph/blob/4166eca33dcb242f5f60262033737867ec8fde59/src/cephadm/cephadm#L6876-L6879

should we avoid having it do so for the binary since we're already deploying it beforehand with `_deploy_cephadm_binary`?

Reply:
wondering if it makes sense to combine this PR with #39619 and avoid this complication